File: sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.entries.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;

public interface BrandService {
    TbBrand findById(Long id);

    List<TbBrand> findAll();

    PageResult<TbBrand> findByPage(int pageNum, int pageSize);

    PageResult<TbBrand> searchByPage(int pageNum, int pageSize, TbBrand searchBrand);

    void add(TbBrand brand);


    void update(TbBrand brand);

    void delete(Long[] ids);
}

File: manager_web/src/main/webapp/js/service/brandService.js
app.service("brandService",function ($http) {
    var baseUrl = "../brand/"
    this.search = function (pageNum,pageSize,searchEntity) {
        if(searchEntity == null){
            return $http.get(baseUrl+"findByPage.do?pageNum="+pageNum+"&pageSize="+pageSize);
        }else{
            return $http.post(baseUrl+"search.do?pageNum="+pageNum+"&pageSize="+pageSize,searchEntity);
        }
    }
    this.insert = function (brand) {
        return $http.post(baseUrl+"add.do",brand);
    }
    this.update = function (brand) {
        return $http.post(baseUrl+"update.do",brand);
    }
    this.findById = function (id) {
        return $http.get(baseUrl+"findById.do?id="+id);
    }
    this.delete = function (ids) {
        return $http.get(baseUrl+"delete.do?ids="+ids);
    }
})